### ISSUE TITLE
Unknown layer: 'FixedDropout'

### ISSUE THREAD

**Issue by gsuni** [Aydinhamedi/Pneumonia-Detection-Ai#150]
Error loading the model: Unknown layer: 'FixedDropout'. Please ensure you are using a `keras.utils.custom_object_scope` and that this object is included in the scope. See https://www.tensorflow.org/guide/keras/save_and_serialize#registering_the_custom_object for details.
this is issue i am facing can you provide custom layer.py file where fixeddropout is defined

**Comment by Aydinhamedi** [comment]
You need to 
import efficientnet.tfkeras
first.

**Comment by Aydinhamedi** [comment]
And if you want the 'FixedDropout' here is a implementation of it in the [original repo](https://github.com/qubvel/efficientnet/blob/master/efficientnet/model.py) with the wrapper 
```python
def get_dropout(**kwargs):
    """Wrapper over custom dropout. Fix problem of ``None`` shape for tf.keras.
    It is not possible to define FixedDropout class as global object,
    because we do not have modules for inheritance at first time.

    Issue:
        https://github.com/tensorflow/tensorflow/issues/30946
    """
    backend, layers, models, keras_utils = get_submodules_from_kwargs(kwargs)

    class FixedDropout(layers.Dropout):
        def _get_noise_shape(self, inputs):
            if self.noise_shape is None:
                return self.noise_shape

            symbolic_shape = backend.shape(inputs)
            noise_shape = [symbolic_shape[axis] if shape is None else shape
                           for axis, shape in enumerate(self.noise_shape)]
            return tuple(noise_shape)

    return FixedDropout
```

**Comment by Aydinhamedi** [comment]
And I added `Utils\FixedDropout.py` for the layer:
```python
from tensorflow.keras import layers, backend

class FixedDropout(layers.Dropout):
    def _get_noise_shape(self, inputs):
        if self.noise_shape is None:
            return self.noise_shape

        symbolic_shape = backend.shape(inputs)
        noise_shape = [symbolic_shape[axis] if shape is None else shape
                       for axis, shape in enumerate(self.noise_shape)]
        return tuple(noise_shape)

    def get_config(self):
        config = super().get_config()
        return config

    @classmethod
    def from_config(cls, config):
        return cls(**config)
```
Example:
```python
from Utils.FixedDropout import FixedDropout
from keras.models import load_model

# Load the model
model = load_model('PAI_model_T.h5', custom_objects={'FixedDropout': FixedDropout})
```

**Comment by Aydinhamedi** [comment]
Can you please confirm if the issue is resolved for you or if you need any further assistance?

**Comment by Aydinhamedi** [comment]
And the only thing that FixedDropout does 
Is fixing a bug in the Dropout layer so you can use the normal Dropout class in new tf.keras versions.

**Comment by gsuni** [comment]
can you please tell me which model should i use  PAI_model_T.h5
and PAI_model_weights in my project .h5.what is difference between them

**Comment by Aydinhamedi** [comment]
`PAI_model_T.h5` file is the full model with the structure and layers but `PAI_model_weights.h5` file only has weights (the parameters).
The weights only file is smaller but you need the [EfficientNet](https://github.com/qubvel/efficientnet) library you can use the full ver `PAI_model_T.h5` and just simply load the model:
```python
from Utils.FixedDropout import FixedDropout
from keras.models import load_model

# Load the model
model = load_model('PAI_model_T.h5', custom_objects={'FixedDropout': FixedDropout})
```
and if you want a smaller model use  [PAI_model_light_T.h5](https://github.com/Aydinhamedi/Pneumonia-Detection-Ai/releases/download/V0.9.4/PAI_model_light_T.h5) the acc is the same but its smaller and lighter.

**Comment by gsuni** [comment]
yes issue is resolved thankyou for replying and resolving my issue

**Comment by gsuni** [comment]
Can you please send code only related to the efficient.net B7.it very
confusing

**Comment by Aydinhamedi** [comment]
Certainly! It seems you’re asking for specific parts of the code related to EfficientNet B7. To clarify, are you looking for the code that involves creating the EfficientNet B7 model, or are you seeking the code for making predictions using a pre-trained EfficientNet B7 model?
And if you want to just try the model download this GUI I made [Link](https://github.com/Aydinhamedi/Pneumonia-Detection-Ai/releases/download/V1.0.3/Interface_GUI.exe)

**Comment by Aydinhamedi** [comment]
And if possible create a new issue about your problem.

**Comment by gsuni** [comment]
It will be good if you provide both the code separately

**Comment by gsuni** [comment]
Code related to the efficient net B7-V6 model in which we get  97.12%

**Comment by Aydinhamedi** [comment]
If you're interested in the code for the model, I'm using this library: [EfficientNet](https://github.com/qubvel/efficientnet)

The code that trains the model is in a Jupyter notebook named `Model_T&T.ipynb`. There are also backups for the codes of older models.

Please note that there are no separate notebooks for each model. This is because the data processing and training parts are the same for every model. In the notebook, there are multiple cells from which you can choose any model.

You can find the code for creating the EfficientNet B7 model in the same notebook.

**Comment by gsuni** [comment]
But it is confusing I am not able to fetch the particular code

**Comment by Aydinhamedi** [comment]
I'm truly sorry but I can`t help you
